src/databaseInterface/src/main/kotlin/org/vaccineimpact/api/db/JooqExtensions.kt

```diff
@@ -2,11 +2,61 @@ package org.vaccineimpact.api.db
 
 import org.jooq.*
 import org.jooq.impl.TableImpl
+import org.postgresql.copy.CopyManager
+import java.io.InputStream
 
 // Just a helper so we can write `fetchInto<T>` instead of `fetchInto(T::class.java)`
 inline fun <reified TRecord : Record> Select<*>.fetchInto(): List<TRecord> = this.fetchInto(TRecord::class.java)
 
 // This helper avoids overloading ambiguity when the field type is "Any"
 fun <T> TableField<*, T>.eqField(otherField: Field<T>): Condition = this.eq(otherField)
 
-fun TableImpl<*>.fieldsAsList() = this.fields().toList()
\ No newline at end of file
+fun TableImpl<*>.fieldsAsList() = this.fields().toList()
+
+fun CopyManager.copyInto(table: TableImpl<*>, stream: InputStream, fields: List<Field<*>>)
+{
+    val fieldList = fields.joinToString(", ") { it.name }
+    val query = "COPY ${table.name} ($fieldList) FROM STDIN"
+    this.copyIn(query, stream)
+}
+
+fun DSLContext.withoutCheckingForeignKeyConstraints(table: TableImpl<*>, work: () -> Unit)
+{
+    val triggers = this.getTriggersFor(table)
```

Comment:
I'm sure its' obvious that it is working from the performance gain but would be nice to test this.

Reply:
Sorry, not sure what you mean. Do you mean you want a test that disabling the triggers is faster?

Reply:
I meant a test to check that the triggers really do get disabled

Reply:
Added!